### DISCUSSION TITLE
logging in jax

### DISCUSSION THREAD

**Discussion by Rasmuskh** [jax-ml/jax#12651] [Q&A] [answered]
Hi, 
I am trying to set up logging for a project and am running into some issues. 

Apparently Jax use of absl somehow [prevents the use of the normal Python logging library](https://github.com/google/jax/discussions/12182), so instead I tried using absl.logging. However, the [documentation for absl](https://abseil.io/docs/python/guides/logging) is extremely sparse and I don't see any way to specify the filepath I want to write the log to.

So my question is how can I write log messages to a chosen filepath when using Jax?

**Comment by froystig** [comment, marked as answer]
If we figure out the remaining bits of #10968, you may no longer need to work with absl logging. We'd welcome your input on that PR!

EDIT: the change is in, as #12769. We no longer require absl logging.

**Reply by Rasmuskh** [reply]
Thanks,
That merge will be a big UX improvement!
However, I need something to work now, preferably without having to work on a bleeding edge release :)
I found this post which solves my immediate problem: https://github.com/abseil/abseil-py/issues/117#issuecomment-982246198

I look forward to not having to use absl.logging in the future! :)

**Reply by froystig** [reply]
The change is in, by the way (as #12769). We no longer require absl logging at head.

**Reply by Rasmuskh** [reply]
Awesome! :)

**Comment by Rasmuskh** [comment]
Pasting Eshijias solution from [here](https://github.com/abseil/abseil-py/issues/117#issuecomment-982246198):
```
import os 
import absl 
from absl import logging

logging.use_absl_handler()

if not os.path.exists('./'): 
     os.makedirs('./') 
logging.get_absl_handler().use_absl_log_file('absl_logging', './') 
absl.flags.FLAGS.mark_as_parsed() 
logging.set_verbosity(logging.INFO)
 
logging.info('test')
```
